packages/core/src/services/db/backends/base-db.provider.ts

```diff
@@ -3,6 +3,12 @@ export type DBValue = string | object | number | boolean
 export type DBResult = DBValue | DBValue[]
 
 export interface DBObject {
+  key: Buffer | string
+  value: DBValue
+}
+
+export interface DBOperation {
+  type: string
```

Comment:
Probably want to use `type: 'put' | 'del'`

Reply:
Oooo that's very cool

Reply:
Hmm so does this mean the type is `a string which equals 'put' or 'del'`? Or does this mean the put or del functions? I was having trouble getting this to work except by using either `string` or `Function`